src/clr-angular/forms/combobox/providers/combobox-focus-handler.service.ts

```diff
@@ -0,0 +1,233 @@
+    });
+  }
+
+  private _trigger: HTMLElement;
+  get trigger() {
+    return this._trigger;
+  }
+  set trigger(el: HTMLElement) {
+    this._trigger = el;
+  }
+
+  private _textInput: HTMLElement;
+  get textInput() {
+    return this._textInput;
+  }
+  set textInput(el: HTMLElement) {
+    this._textInput = el;
+    this.renderer.setAttribute(el, 'id', this.id);
+    this.renderer.listen(el, 'keydown', event => !this.handleTextInput(event));
+    this.renderer.listen(el, 'blur', event => {
+      // event.relatedTarget is null in IE11. In that case we use document.activeElement
+      // which points to the element that becomes active as the blur event occurs on the input.
+      const target = event.relatedTarget || document.activeElement;
+
+      if (target && isPlatformBrowser(this.platformId)) {
+        // if for some reason the user clicks on the trigger while navigating with keyboard,
+        // we mark it to move focus back on the input for later when it is closed.
+        if (target === this.trigger) {
+          this.focusBackOnTextInput = true;
+          return;
+        }
+        // if we are focusing on something inside container keep it open
+        if (this.container && this.container.contains(target)) {
+          return;
+        }
+        // in browsers other than IE11, the target points to BODY
+        // if we are focusing on something inside container
+        if (target.tagName === 'BODY') {
+          return;
+        }
+      }
+
+      // we clear out any existing focus on the items
+      this.children.pipe(take(1)).subscribe(items => items.forEach(item => item.blur()));
+
+      // We let the user move focus to where the want, we don't force the focus back on the textInput
+      this.focusBackOnTextInput = false;
+      this.toggleService.open = false;
+    });
+  }
+
+  private moveFocusTo(direction: ArrowKeyDirection) {
+    // todo: remove following todos and comments after verifying it works in various browsers
+    // todo: check why the timeout
+    // looks like it works without it fine
+    // setTimeout(() => {
+    this.focusService.moveTo(this);
+    this.focusService.move(direction);
+    // todo: check why it is here and remove
+    // this breaks focusing on 1st element on open
+    // if (this.container && isPlatformBrowser(this.platformId)) {
+    //   this.container.focus();
+    // }
+    // });
+  }
+
+  private openAndMoveTo(direction: ArrowKeyDirection) {
+    if (!this.toggleService.open) {
+      this.toggleService.openChange.pipe(take(1)).subscribe(open => {
+        if (open) {
+          this.moveFocusTo(direction);
+        }
+      });
+      this.toggleService.open = true;
+    } else {
+      this.moveFocusTo(direction);
+    }
+  }
+
+  // this service is only interested in keys that may move the focus
+  private handleTextInput(event) {
+    let preventDefault = false;
+    if (event) {
+      switch (event.keyCode) {
+        case ENTER:
```

Comment:
I can tell why it happens:
Because as you hit enter and type, no actual selection is happening. Only filtering. You need to go to the list of options (even if only 1 option remains) and make a selection there.
I also don't like that if you type, go to the list and return, your previously typed letters are discarded. It's not unusable, but it's not very usable too. I'm leaving it open for discussion and thinking a bit more on it. The option, as I see them are like this:
- If only one option remains after filtering, allow selecting it with a single Enter-keypress.
- Try to keep the text if you go back form the list to the input.

Reply:
Looking again at the W3C examples it should close on enter, but not clear the currently typed letters. We can improve on this... I'll handle it after finishing the rest of the review items, as it's severity seems rather small for its scope to me.

Reply:
Unresolved it, so I can keep track of it. May spawn a low priority issue and will re-resolve it again.

Reply:
Updated to W3C examples with last commit.